packages/emotion-theming/types-tests-global-theme/index.d.ts

```diff
@@ -13,37 +11,27 @@ import {
 } from '@emotion/styled'
 import { PropsOf, DistributiveOmit } from '@emotion/styled-base'
 
-export interface ThemeProviderProps<Theme> {
-  theme: Partial<Theme> | ((outerTheme: Theme) => Theme)
+export interface ThemeProviderProps {
+  theme: Partial<Emotion.Theme> | ((outerTheme: Emotion.Theme) => Emotion.Theme)
```

Comment:
Hmm... actually, @Andarist , what is the reasoning behind `Partial` here? What's the point of a typed theme, if the Provider can ignore it? These typings produce errorprone code, but I see this has always been the case.

With errorprone, I mean, since this asks for a partial, one can provide incomplete theme here, but everywhere else, a concrete Theme definition is being used.

So, I could have a Theme:
```
interface Theme {
  colors: {
    someNamespace: {
      red: string
      blue: string
    }
  }
}
```

But the theme provider allows for:
```
<ThemeProvider theme={{}} />
```

While the usage is typed against full theme and allows:
```
styled.div`
  color: ${props => props.theme.colors.namespace.red}
`
```

Reply:
That is there because you can nest providers and inner theme gets spread onto the outer one:
https://github.com/joltmode/emotion/blob/next/packages/emotion-theming/src/theme-provider.js#L30

Not sure how to accommodate for both things - forcing top level to receive all, but partials for nested ones. The only thing which come to my mind is aliasing and providing different types for both

Reply:
Aha, makes sense. 
Well, yeah, contextual typing would be great, but that's one hell of a task!

In any way, I guess that's usecase specific then and the library shouldn't really tackle that deep.

Should the user want to retype and alias that's up to them. Buuut, the overall concept of this PR is to get rid of any theme retyping, so that would make it impossible. Hence why I asked a general question on this PR main conversation...

Have all implications been evaluated on removing Theme parameter and hardcoding to global one?

It's fine by me, haven't yet found a usecase for two ThemeProviders.

Reply:
We use this a fair bit actually, our website changes part of the theme depending on the section.

So we provide a bunch of global metrics, then have 2 more theme providers a bit deeper in our render tree which provide the other properties. With the 3 ThemeProviders we have specified all the props on our `theme`. I wouldn't expect TypeScript to save me if I happened to accidentally remove one of them